flink-clients/src/main/java/org/apache/flink/client/program/StreamContextEnvironment.java

```diff
@@ -161,6 +189,16 @@ public static void setAsContext(
             final boolean suppressSysout) {
         StreamExecutionEnvironmentFactory factory =
                 conf -> {
+                    final List<JobValidationError> errors = new ArrayList<>();
+                    final boolean allowConfigurations =
+                            configuration.getBoolean(
+                                    DeploymentOptions.ALLOW_CLIENT_JOB_CONFIGURATIONS);
+                    if (!allowConfigurations && !conf.toMap().isEmpty()) {
+                        conf.toMap()
+                                .forEach(
+                                        (k, v) ->
+                                                errors.add(new ConfigurationNotAllowedError(k, v)));
```

Comment:
do we really need to create a new ConfigurationNotAllowedError instance for each kv config? With only one ConfigurationNotAllowedError could already make sure the errors is not empty.

Reply:
As a user, I would like to see all the configurations I am currently using which are not allowed. If we only show a generic message it might be hard to find the cause.
Is this what you mean?

Reply:
yes, I thought the requirement was no configuration will be allowed by the user at all. If this is right, showing error messages for every config kvs might become to be noises to the user who would need more effort to walk through all these errors and finally found that none customer config is allowed.